compiler/luci/service/src/Nodes/CircleFullyConnected.cpp

```diff
@@ -36,4 +40,76 @@ luci::CircleNode *CloneNodeLet<CN::DEF>::visit(const luci::CircleFullyConnected
   return cloned;
 }
 
+namespace sinf
+{
+
+loco::TensorShape Algorithm::visit(const luci::CircleFullyConnected *node)
+{
+  auto input_shape = circle_shape(loco::must_cast<CircleNode *>(node->input()));
+  auto weights_shape = circle_shape(loco::must_cast<CircleNode *>(node->weights()));
+
+  loco::TensorShape out_shape;
+
+  // NOTE Some recipes in some repositories are using rank 4 input for FullyConnected.
+  //      Until they are all fixed, disable following assert.
+  // TODO Enable following assert after related fixes are applied
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L194
+  // LUCI_ASSERT(input_shape.rank() == 2 || input_shape.rank() == 3,
+  //             "Input rank of FullyConnected should be 2 or 3");
+
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L225
+  LUCI_ASSERT(weights_shape.rank() == 2, "Weights of FullyConnected should be 2");
+
+  // https://github.com/tensorflow/tensorflow/blob/ea33c1e7a25d8025e8ee405ad8ab7be261798d76/tensorflow/lite/kernels/fully_connected.cc#L353-L367
+  if (node->keep_num_dims())
+  {
+    // [1,10,?,20] X [15,20] -> [1,10,?,15] (weights_shape =[15,20])
```

Comment:
Can you describe in some words?

Reply:
Would this change be okay?
```suggestion
    // [1,10,?,20] X [15,20] -> [1,10,?,15]
    // When Input_shape is [1,10,?,20] and weights_shape is [15,20], with keep_num_dims set to true, 
    // the FullyConnected operation works when the last dimension of input_shape matches the last dimension of weights_shape. 
    // In this case, the dimensions of input_shape are retained, 
    //and the number of outputs (n_output) is set to match the first dimension of weights_shape.
```

Reply:
```suggestion
```

The section under the condition `if (node->keep_num_dims())` appears to be directly moved from the existing code without any modifications. (from [CircleShapeInferenceRule.cpp](https://github.com/Samsung/ONE/pull/13875/files#diff-d9e5543aaa75f5b71226c51b9348297cbf395eb02fcced4be04c8fb81446e30d))

It seems that there wasn't an example comment included before.
Could you please clarify if there is a specific reason for adding this comment here?

If it's the same part as the old code, how about just not adding this comment?

Reply:
I agree, It just moved from the exiting code.
So, I don't think it's necessary to add comment.